src/radical/saga/adaptors/shell/shell_job.py

```diff
@@ -622,6 +655,19 @@ def initialize(self):
         self._logger.debug("got mon prompt(%s)(%s)" % (ret, out.strip()))
 
 
+    # ----------------------------------------------------------------
+    #
+    def _ping_cb (self) :
+
+        with self._shell_lock:
+
+            _, out, _ = self.shell.run_sync('PING')
+            assert('PONG' in out), out
+
+            self._ping = mt.Timer(_PING_DELAY, self._ping_cb)
+            self._ping.start()
```

Comment:
these 2 lines are repeated, `self._ping` was set in `initialization` method (L526)

Reply:
That's on purpose: the `self_ping` timer can only be called once - so whenever it is called, it re-initializes a new timer instance for the next call.  I don't like it either, but that's how the threading.Timer class works _*shrug*_